File: src/BaseHmiTypes/Screens/Base/HmiOcxControl.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiOcxControl : HmiCustomWidgetContainer
{
    public HmiOcxControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiOcxControl;
    }

    public string? OcxGuid { get; set; }

    public string? OcxName { get; set; }

    public string? OcxProgramId { get; set; }

    public string? OcxFileName { get; set; }

    public string? OcxFileVersion { get; set; }

    public string? OcxTypeLibrary { get; set; }

    public string? OcxTypeLibraryVersion { get; set; }

    public byte[]? OcxState { get; set; }

    public string? OcxStateFormat { get; set; }

    /// <summary>
    /// XML document embedded in <see cref="OcxState"/>, when the source format exposes one.
    /// The original state bytes remain authoritative and are preserved unchanged.
    /// </summary>
    public string? OcxStateXml { get; set; }

    public HmiAxHostState? AxHostState { get; set; }
}

File: typescript/base-hmi-types/src/screens/base/HmiOcxControl.ts
import { HmiCustomWidgetContainer } from "./HmiCustomWidgetContainer.js";
import { HmiAxHostState } from "./HmiAxHostState.js";
import { HmiObjectType } from "./HmiObjectType.js";

export class HmiOcxControl extends HmiCustomWidgetContainer {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiOcxControl;
  }

  ocxGuid?: string;
  ocxName?: string;
  ocxProgramId?: string;
  ocxFileName?: string;
  ocxFileVersion?: string;
  ocxTypeLibrary?: string;
  ocxTypeLibraryVersion?: string;
  ocxState?: Uint8Array;
  ocxStateFormat?: string;
  axHostState?: HmiAxHostState;
}
